src/pybind/ceph_volume_client.py

```diff
@@ -633,10 +634,11 @@ def create_volume(self, volume_path, size=None, data_isolated=False):
                 })
             self.fs.setxattr(path, 'ceph.dir.layout.pool', pool_name, 0)
 
-        # enforce security isolation, use seperate namespace for this volume
-        namespace = "{0}{1}".format(self.pool_ns_prefix, volume_path.volume_id)
-        log.info("create_volume: {0}, using rados namespace {1} to isolate data.".format(volume_path, namespace))
-        self.fs.setxattr(path, 'ceph.dir.layout.pool_namespace', namespace, 0)
+        # enforce security isolation, use separate namespace for this volume
+        if namespace_isolated:
+            namespace = "{0}{1}".format(self.pool_ns_prefix, volume_path.volume_id)
+            log.info("create_volume: {0}, using rados namespace {1} to isolate data.".format(volume_path, namespace))
+            self.fs.setxattr(path, 'ceph.dir.layout.pool_namespace', namespace, 0)
```

Comment:
@ajarr If namespace is isolated, we set `ceph.dir.layout.pool_namespace` on path, which will also set `ceph.dir.layout.pool` (inherited from ancestor) on it. If namespace is not isolated, should we configure `ceph.dir.layout.pool` on volume path intentionally? I'm afraid if ancestor pool changed, we may have permission issues.

Reply:
@cofyc , I looked at https://github.com/kubernetes-incubator/external-storage/pull/674/commits/29131951c375a8f9c8d4a03e3ebbb3a485e67175
makes sense. thanks!

Reply:
@cofyc done.